locked-asset/token-unstake/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+#![no_std]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+pub const DEFAULT_UNBOND_EPOCHS: u64 = 10;
+
+#[derive(ManagedVecItem, TopEncode, TopDecode, NestedEncode, NestedDecode, TypeAbi, Clone)]
+pub struct UnstakePair<M: ManagedTypeApi> {
+    pub unlock_epoch: u64,
+    pub token_payment: EsdtTokenPayment<M>,
+}
+
+#[elrond_wasm::contract]
+pub trait TokenUnstakeModule: token_send::TokenSendModule {
+    #[init]
+    fn init(&self) {
```

Comment:
Since this is now a separate contract (and we don't pollute the already heavy `init` of `energy-factory`, I'd say we can add `unbond_epochs` as init argument, and remove the default const.

Reply:
Additionally, as we discussed, let's add a `WhitelistMapper` for whitelisted tokens, and give a `MultiValueEncoded` of `TokenIdentifier` as argument here. Most likely this will be for one token only initially, but maybe we could reuse this for multiple different contracts.